### PR TITLE
Support multiple checksum algorithms

### LINKED ISSUE #146: Support new checksum algorithm (sha256)
BBB is about to switch form SHA1 to SHA256 as the checksum algorithm (see bigbluebutton/bigbluebutton#15684).

Hence, this library should allow its clients to specify which algorithm to use.

https://github.com/littleredbutton/bigbluebutton-api-php/blob/98f1a0cf330b2b18f928d9e45f4379506d5b3332/src/Util/UrlBuilder.php#L59-L69

### PR DESCRIPTION
**Pull request by SamuelWei**
Fixes #146 

**Changes**
- Adding constructor argument to set the hash algorithm
-  Adjust tests

### CODE REVIEW

**Review by sualko — APPROVED** [review]
Looks good.

**Comment by SamuelWei** [comment]
> Looks good.

@sualko Can you just have a quick look into the linked issue, as I started a question on how to deal with it. This is just one implementation idea

File: src/BigBlueButton.php [hunk 1]
```diff
@@ -108,17 +113,19 @@ class BigBlueButton
      *
      * @throws ConfigException
      */
-    public function __construct(?string $baseUrl = null, ?string $secret = null, ?TransportInterface $transport = null)
+    public function __construct(?string $baseUrl = null, ?string $secret = null, ?TransportInterface $transport = null, ?string $hashAlgorithm = 'sha1')
```
Comment (FelixJacobi) [inline src/BigBlueButton.php:116]:
```suggestion
    public function __construct(?string $baseUrl = null, ?string $secret = null, ?TransportInterface $transport = null, string $hashAlgorithm = 'sha1')
```

It does not make sense to allow `null` here.

**Review by FelixJacobi — CHANGES_REQUESTED** [review]

**Comment by SamuelWei** [comment]
Changed the initial implementation using a 4th constructor argument to a fluent setter to be compliant with the official PHP API

**Comment by FelixJacobi** [comment]
> Changed the initial implementation using a 4th constructor argument to a fluent setter to be compliant with the official PHP API

Hm, I was fine with the initial implementation,  except the type hint 😄  . In my opinion, setters, which changes the behavior of the object are bad at all. Imagine the following use case: A developer to knowing the internals of this library injects the object using dependency injection in his own class and just thinks it's a good idea to just set the algorithm. This changes the state aferwards permanently and globally. As the algorithm is a connection parameter like the URL and secret, I would not allow changing it after setting it in the constructor.

**Comment by SamuelWei** [comment]
@FelixJacobi I reverted it back to the original initial implementation

**Review by FelixJacobi — APPROVED** [review]

**Pull request merged**